### PR TITLE
Allow a custom header name for authorization in Prometheus

### PR DESCRIPTION
**Pull request by amitkharb**
## Additional context and related issues

Allowing users to use a custom header for authorization of prometheus endpoint

Current code only support Authorization http header, just adding a new configuration property to allow a custom header name

## Release notes
(X ) Release notes are required, with the following suggested text:

```markdown
# Prometheus
* Add support for a custom authorization header name.({issue}`21187`)
```

### CODE REVIEW

**Comment by amitkharb** [comment]
All the changes from https://github.com/trinodb/trino/pull/20939 has been applied here. 
I could not get the other PR cla-bot  signed.

**Review by ebyhr — APPROVED** [review]

File: docs/src/main/sphinx/connector/prometheus.md [hunk 1]
```diff
@@ -42,13 +42,14 @@ prometheus.read-timeout=10s
 The following configuration properties are available:
 
 | Property name                               | Description                                                                                  |
-| ------------------------------------------- | -------------------------------------------------------------------------------------------- |
+|---------------------------------------------|----------------------------------------------------------------------------------------------|
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/prometheus.md:45]: Revert unrelated change.

Comment (amitkharb) [reply docs/src/main/sphinx/connector/prometheus.md]: DONE

**Review by mosabua — APPROVED** [review]
Looks good
